src/mds/Migrator.cc

```diff
@@ -2593,6 +2582,14 @@ void Migrator::import_remove_pins(CDir *dir, set<CDir*>& bounds)
   }
 }
 
+class C_MDC_QueueContext : public MigratorContext {
+public:
+  std::list<MDSInternalContextBase*> contexts;
+  C_MDC_QueueContext(Migrator *m) : MigratorContext(m) {}
+  void finish(int r) override {
+    get_mds()->queue_waiters_front(contexts);
```

Comment:
Why do these need queued at the front?

Reply:
This will require that `MDSRank::finished_queue` be a `std::deque` which is not preferred to `std::vector` (in #23195 ). So, is it actually important to queue at the front?

Reply:
yes, it's important.  to make sure contexts get processed in proper order

Reply:
Please document an example.